### PR TITLE
Add publisher for VirtualServer

### PR DESCRIPTION
**Pull request by ar4s**

### CODE REVIEW

File: src/ralph/ralph2_sync/publishers.py [hunk 1]
```diff
@@ -183,3 +227,28 @@ def sync_configuration_class_to_ralph2(sender, instance=None, created=False, **k
...
+        'ralph2_parent_id': _get_obj_id_ralph_2(instance.parent) if instance.parent else None,  # noqa
+        'hostname': instance.hostname,
+        'sn': instance.sn,
+        'type': instance.type.name,
+        'status': virtual_server_type_asset_mapping[instance.status],
+        'service_uid': instance.service_env.service.uid,
+        'environment_id': _get_obj_id_ralph_2(
+            instance.service_env.environment
+        ),
+        'venture_role_id': venture_role_id
```
Comment (mkurek) [inline src/ralph/ralph2_sync/publishers.py:228]: venture would be nice here as well, just like in DCAsset

File: src/ralph/ralph2_sync/publishers.py [hunk 2]
```diff
@@ -183,3 +227,28 @@ def sync_configuration_class_to_ralph2(sender, instance=None, created=False, **k
...
+    )
+    return {
+        'id': instance.id,
+        'ralph2_id': _get_obj_id_ralph_2(instance),
+        'ralph2_parent_id': _get_obj_id_ralph_2(instance.parent) if instance.parent else None,  # noqa
+        'hostname': instance.hostname,
+        'sn': instance.sn,
+        'type': instance.type.name,
+        'status': virtual_server_type_asset_mapping[instance.status],
+        'service_uid': instance.service_env.service.uid,
```
Comment (mkurek) [inline src/ralph/ralph2_sync/publishers.py]: what if service is empty?

File: src/ralph/ralph2_sync/publishers.py [hunk 3]
```diff
@@ -183,3 +227,28 @@ def sync_configuration_class_to_ralph2(sender, instance=None, created=False, **k
...
+        instance.configuration_path
+    )
+    return {
+        'id': instance.id,
+        'ralph2_id': _get_obj_id_ralph_2(instance),
+        'ralph2_parent_id': _get_obj_id_ralph_2(instance.parent) if instance.parent else None,  # noqa
+        'hostname': instance.hostname,
+        'sn': instance.sn,
+        'type': instance.type.name,
+        'status': virtual_server_type_asset_mapping[instance.status],
```
Comment (mkurek) [inline src/ralph/ralph2_sync/publishers.py]: ralph2 doesn't use status field - remove it

File: src/ralph/ralph2_sync/publishers.py [hunk 4]
```diff
@@ -183,3 +204,28 @@ def sync_configuration_class_to_ralph2(sender, instance=None, created=False, **k
...
+    return {
+        'id': instance.id,
+        'ralph2_id': _get_obj_id_ralph_2(instance),
+        'ralph2_parent_id': _get_obj_id_ralph_2(instance.parent) if instance.parent else None,  # noqa
+        'hostname': instance.hostname,
+        'sn': instance.sn,
+        'type': instance.type.name,
+        'service_uid': instance.service_env.service.uid if instance.service_env else None,  # noqa
+        'environment_id': _get_obj_id_ralph_2(
+            instance.service_env.environment
```
Comment (mkurek) [inline src/ralph/ralph2_sync/publishers.py:225]: `if instance.service_env`

Comment (ar4s) [inline src/ralph/ralph2_sync/publishers.py:225]: done

File: src/ralph/ralph2_sync/publishers.py [hunk 5]
```diff
@@ -17,10 +17,29 @@
...
+    venture_role_id = _get_obj_id_ralph_2(configuration_path)
+    if venture_id is None:
+        logger.error('ConfigurationModule {} not found when syncing'.format(
+            configuration_path.module.id
+        ))
+    if venture_role_id is None:
+        logger.error('ConfigurationClass {} not found when syncing'.format(
+            configuration_path.id
+        ))
+        configuration_path.module
```
Comment (mkurek) [inline src/ralph/ralph2_sync/publishers.py]: rm it

**Comment by mkurek** [comment]
:ok:

**Pull request merged**